src/ejabberd_users.erl

```diff
@@ -50,13 +50,15 @@ start(Host) ->
      5000,
      worker,
      [ejabberd_users]},
+    %% XXX lifecycle of this child is broken
```

Comment:
can you elaborate more on that?

Reply:
it's not host specific. So, start/stop procedure would keep old data.
But otherwise we would have a table per host, which is not optimal too.

Reply:
ah, waaait.
It's host specific. But we don't check if it's already running

Reply:
So, I need mongoose_service here

Reply:
hmm, gen_mod deps support is broken, but would be required to convert this file into a module.
I am happy with this code for now, I don't want to change starting logic here YET.